### PR TITLE
$graphLookup Support (initial)

### PR DESCRIPTION
**Pull request by jed204**
I'm not happy with the 'startWith' field as it's not being properly used (yet). Is there a similar usage within the project to this field? I didn't notice one on some searches but I may have missed it.

More test cases to be developed but it's saying the output doesn't matching but the assert output suggests they are? Any insight here would be great.

```
Expecting:
  <[{"manager"="Dave", "managers"=[{"_id"=1, "name"="Mike", "order"=1},
    {"_id"=2, "manager"="Mike", "name"="Dave", "order"=0}], "name"="Bob"}]>
to contain exactly (and in same order):
  <[{"manager"="Dave", "managers"=[{"_id"=1, "name"="Mike", "order"=1},
    {"_id"=2, "manager"="Mike", "name"="Dave", "order"=0}], "name"="Bob"}]>
but some elements were not found:
  <[{"manager"="Dave", "managers"=[{"_id"=1, "name"="Mike", "order"=1},
    {"_id"=2, "manager"="Mike", "name"="Dave", "order"=0}], "name"="Bob"}]>
and others were not expected:
  <[{"manager"="Dave", "managers"=[{"_id"=1, "name"="Mike", "order"=1},
    {"_id"=2, "manager"="Mike", "name"="Dave", "order"=0}], "name"="Bob"}]>
```

### CODE REVIEW

File: core/src/main/java/de/bwaldvogel/mongo/backend/aggregation/Aggregation.java [hunk 1]
```diff
@@ -12,20 +12,7 @@
...
-import de.bwaldvogel.mongo.backend.aggregation.stage.LimitStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.LookupStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.LookupWithPipelineStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.MatchStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.OrderByStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.ProjectStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.ReplaceRootStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.SkipStage;
-import de.bwaldvogel.mongo.backend.aggregation.stage.UnwindStage;
+import de.bwaldvogel.mongo.backend.aggregation.stage.*;
```
Comment (markbigler) [inline core/src/main/java/de/bwaldvogel/mongo/backend/aggregation/Aggregation.java]: Using wildcard import statements is not a good idea for several reasons. You can find some explanaitions here: https://stackoverflow.com/a/147461

Comment (jed204) [reply core/src/main/java/de/bwaldvogel/mongo/backend/aggregation/Aggregation.java]: Agreed - the IDE ended up doing that, didn't notice - thanks.

**Comment by bwaldvogel** [comment]
Thanks for your contribution. I would be happy to merge once all tests pass ;)
Could you also add one or two integration tests in `de.bwaldvogel.mongo.backend.AbstractAggregationTest`? With this test class we can make sure that the implementation in `mongo-java-server` behaves the same as a real MongoDB…

File: core/src/main/java/de/bwaldvogel/mongo/backend/aggregation/Aggregation.java (hunk 1 above)
Comment (bwaldvogel) [reply core/src/main/java/de/bwaldvogel/mongo/backend/aggregation/Aggregation.java]: If you use IDEA or Eclipse, you could try to import the code style from [/code-style](https://github.com/bwaldvogel/mongo-java-server/tree/master/code-style)

**Comment by zpapez** [comment]
@bwaldvogel I've forked @jed204 's branch and tried to finish the graphLookup functionality with the `de.bwaldvogel.mongo.backend.AbstractAggregationTest` that you mentioned.
I opened it as PR https://github.com/bwaldvogel/mongo-java-server/pull/122

**Comment by bwaldvogel** [comment]
With PR #122 we do have initial support for `$graphLookup` on master (thanks @zpapez, @jed204 and @markbigler!).

@jed204: Could you have a look if your test cases are successful?

**Pull request closed without merging**